# One Sample Wilcoxon Non Parametric Hypothesis Test

non parametric hypothesis test is a rank-based test and it compares the standard value (theoretical value) with the hypothesized median\




### Hypothesis of 1 sample Wilcoxon Signed test

Left-tailed test:

- Null Hypothesis H0: The population median (η) is greater than or equal to hypothesized median(η0)- η≥ η0
- Alternative Hypothesis : H1: The population median(η) is less than the hypothesized median(η0) – η<η0

Right-tailed test:

- Null Hypothesis H0: The population median (η) is less than or equal to hypothesized median(η0)- η≤η0
- Alternative Hypothesis : H1: The population median(η) is greater than the hypothesized median(η0)- η>η0

Two-tailed test:

- Null Hypothesis H0: The population median (η) is equal to hypothesized median(η0)- η=η0
- Alternative Hypothesis : H1: The population median(η) is not equal to the hypothesized median(η0)- η≠η0


### Assumptions of the one sample Wilcoxon test

- Differences between the data value and the hypothesized median are continuous
- Data follows the symmetric distribution
- Observations are mutually independent of each other
- Measurement scale is at least an interval

---

## Example 1 - Manual (Understanding One Sample Wilcoxon test)

random samples of 10 students’ marks are noted below, is there evidence at the 5% confidence level to suggest that the median mark is greater than 67?

![sample](./Images/sample_wilcoxon1.png)

Right tailed Hypothesis:

- Null Hypothesis H0: The population median value ≤ 67 marks
- Alternative Hypothesis : H1: The population median value > 67 marks

NOTE
>Ignoring the signs, rank the differences smallest rank =1

![sample](./Images/sample_wilcoxon2.png)

Separate the positive and negative ranks

- Sum of + ranks =40
- Sum of – ranks=15
- Smallest value among 40 & 15 =15
- Test statistics =15
- Critical value =11

Criteria:
>The null hypothesis will be rejected if the test statistic, W, is less than or equal to the critical value.

![sample](./Images/sample_wilcoxon4.png)

Conclusion:
>Since the test statistic value is greater than the critical value, hence we fail to reject the null hypothesis. There is no significant evidence that the median rank is greater than 67.

---

## Example 2 Python

a small sample of 21 numbers: 4, 5, 6, 5, 3, 4, 2, 7, 6, 5, 4, 3, 8, 8, 9, 4, 6, 7, 2, 5, 6. Null hypothesis, median equals 5; alternative, it doesn't. Alpha is 0.05 and not null differences (between sample elements and hypothetical median) are 17.

In [2]:
import numpy as np
from scipy.stats import wilcoxon as wil

sample = [4, 5, 6, 5, 3, 4, 2, 7, 6, 5, 4, 3, 8, 8, 9, 4, 6, 7, 2, 5, 6]
sample = np.array(sample)
med0 = 5 # median of the sample

W, p = wil(sample - med0, correction = False, mode = 'approx')

print("W = %.1f, p-value = %.3f" % (W, p))

W = 68.0, p-value = 0.683


Conclusion 

>get 68 for W- and 85 for W+. According to my notes, I have to choose the minimum of both, so 68, and it might be bigger than the critical value from the table, W17,0.05, which it seems to be 41.

---

# Wilcoxon Signed-Rank Test in Python

The Wilcoxon Signed-Rank Test is the non-parametric version of the paired samples t-test.

It is used to test whether or not there is a significant difference between two population means when the distribution of the differences between the two samples cannot be assumed to be normal.

## Example 1

Case Statement
>
>Researchers want to know if a new fuel treatment leads to a change in the average mpg of a certain car. To test this, they measure the mpg of 12 cars with and without the fuel treatment.
>
>Use the following steps to perform a Wilcoxon Signed-Rank Test in Python to determine if there is a difference in the mean mpg between the two groups.


In [5]:
import scipy.stats as stats

group1 = [20, 23, 21, 25, 18, 17, 18, 24, 20, 24, 23, 19]
group2 = [24, 25, 21, 22, 23, 18, 17, 28, 24, 27, 21, 23]



#perform the Wilcoxon-Signed Rank Test
w, p = stats.wilcoxon(group1, group2, correction=False)
print("W = %.1f, p-value = %.3f" % (W, p))

W = 68.0, p-value = 0.044


Findings
>The test statistic is 10.5 and the corresponding two-sided p-value is 0.044.

Results Interpretation
>
>In this example, the Wilcoxon Signed-Rank Test uses the following null and alternative hypotheses:
>
>- H0: The mpg is equal between the two groups
>
>- HA: The mpg is not equal between the two groups
>
Since the p-value (0.044) is less than 0.05, we reject the null hypothesis. We have sufficient evidence to say that the true mean mpg is not equal between the two groups.